arch/arch.go

```diff
@@ -62,11 +63,58 @@ func ubuntuArchFromGoArch(goarch string) string {
 		"ppc64le": "ppc64el",
 		"s390x":   "s390x",
 		"ppc":     "powerpc",
+		// available in debian and other distros
+		"ppc64": "ppc64",
 	}
 
 	ubuntuArch := goArchMapping[goarch]
 	if ubuntuArch == "" {
-		log.Panicf("unknown goarch %v", goarch)
+		log.Panicf("unknown goarch %q", goarch)
+	}
+
+	return ubuntuArch
+}
+
+// UbuntuKernelArchitecture return the debian equivalent architecture
+// for the current running kernel. This is usually the same as the
+// UbuntuArchitecture - however there maybe cases that you run e.g.
+// a snapd:i386 on an amd64 kernel.
+func UbuntuKernelArchitecture() string {
+	var utsname syscall.Utsname
+	if err := syscall.Uname(&utsname); err != nil {
+		log.Panicf("cannot get kernel architecture: %v", err)
+	}
+
+	kernelArch := make([]byte, 0, len(utsname.Machine))
```

Comment:
utsname.Machine is defined as `[65]int8` so doing the above is slightly more involved. The following will work:
```
	p := (*[len(utsname.Machine)]byte)(unsafe.Pointer(&utsname.Machine))
	kernelArch := bytes.SplitN(p[:], []byte("\x00"), 2)[0]
```
If you prefer that, I'm happy to use that instead.